one-ui/src/main/java/com/marklogic/hub/oneui/auth/AuthenticationFilter.java

```diff
@@ -137,17 +141,38 @@ protected AuthenticationToken authenticateUser(String username, String password)
      * - see https://docs.spring.io/spring-security/site/docs/current/reference/htmlsingle/#appendix-faq-role-prefix.
      *
      * @param stagingClient
+     * @param roles
+     * @param canInstallDataHub
      * @return
      */
-    protected List<GrantedAuthority> getAuthoritiesForAuthenticatedUser(DatabaseClient stagingClient) {
-        List<GrantedAuthority> authorities = new ArrayList<>();
+    protected Pair<List<GrantedAuthority>, ArrayNode> getAuthoritiesForAuthenticatedUser(DatabaseClient stagingClient, ArrayNode[] roles, boolean[] canInstallDataHub) {
+        List<GrantedAuthority> grantAuthorities = new ArrayList<>();
+        ArrayNode authorities = null;
         JsonNode response = SecurityService.on(stagingClient).getAuthorities();
-        if (response != null && response.has("authorities")) {
-            response.get("authorities").iterator().forEachRemaining(node -> {
-                authorities.add(new SimpleGrantedAuthority("ROLE_" + node.asText()));
-            });
+        if (response != null) {
+            if (response.has("authorities")) {
+                response.get("authorities").iterator().forEachRemaining(node -> {
+                    String authority = node.asText();
+                    if ("canInstallDataHub".equals(authority))  {
```

Comment:
Instead of this special case for this one capability, just add a method to AuthenticationToken like this:

- boolean hasAuthority(String authorityName);

LoginFilter can then just call token.hasAuthority("canInstallDataHub")

Then you don't need the boolean array of a single value.

Reply:
@rjrudin I thought about using another method before,.But as we discussed, we do not need to have hasManagerPrivileges after UI change. And I will change the method to remove the canInstallDataHub array.

Reply:
Okay, that's fair.